### ISSUE TITLE
what is the purpose of declaring it outside?

### ISSUE THREAD

**Issue by Prashanith** [coding-Studio-vbit/ReliefRiders#181]
what is the purpose of declaring it outside?

_Originally posted by @abhiram2600 in https://github.com/coding-Studio-vbit/ReliefRiders/pull/178#discussion_r714006178_

This was buggy, it was easy doing it this way

**Comment by abhiram2600** [comment]
It didn't really answer my question though XD. Anyway, you cannot say it was buggy and have a workaround for something so trivial. Maybe investigate more why whatever buggy thing that happened was occurring.

**Comment by Prashanith** [comment]
This screen had several bugs modularized so it could be easy to debug

**Comment by abhiram2600** [comment]
so does that mean changing it back inside the loop still works without any bugs? if it does, then change it back, or else it itself is a bug you might need to investigate further.

**Comment by Prashanith** [comment]
Yeah will do that and the page is not complete yet so I requested SaiKiran not to merge the PR

**Comment by abhiram2600** [comment]
cool. Add a (WIP) in the title end of the PR if it isn't complete. WIP means work in progress.

**Comment by Prashanith** [comment]
nice one will update it

**Comment by Revantark** [comment]
If updated close this.

**Comment by Prashanith** [comment]
> what is the purpose of declaring it outside?
> 
> _Originally posted by @abhiram2600 in [#178 (comment)](https://github.com/coding-Studio-vbit/ReliefRiders/pull/178#discussion_r714006178)_
> 
> This was buggy, it was easy doing it this way

Fixed this. I separated into several functions to identify the bug easily.  Now it looks clean
